elements/power.lua

```diff
@@ -160,58 +161,72 @@ local Update = function(self, event, unit)
 
 	power.disconnected = disconnected
 
-	local r, g, b, t
-
-	if(power.colorTapping and not UnitPlayerControlled(unit) and UnitIsTapDenied(unit)) then
-		t = self.colors.tapped
-	elseif(power.colorDisconnected and disconnected) then
-		t = self.colors.disconnected
-	elseif(displayType == ALTERNATE_POWER_INDEX and power.altPowerColor) then
-		t = power.altPowerColor
-	elseif(power.colorPower) then
-		local ptype, ptoken, altR, altG, altB = UnitPowerType(unit)
-
-		t = self.colors.power[ptoken]
-		if(not t) then
-			if(power.GetAlternativeColor) then
-				r, g, b = power:GetAlternativeColor(unit, ptype, ptoken, altR, altG, altB)
-			elseif(altR) then
-				-- As of 7.0.3, altR, altG, altB may be in 0-1 or 0-255 range.
-				if(altR > 1) or (altG > 1) or (altB > 1) then
-					r, g, b = altR / 255, altG / 255, altB / 255
+	local ptype, ptoken, altR, altG, altB = UnitPowerType(unit)
+	local t = self.colors.power[ptoken]
+	local r, g, b
+	local atlas = power.atlas or t.atlas
+	local bg = power.bg
+
+	if(power.useAtlas and atlas and displayType ~= ALTERNATE_POWER_INDEX) then
+		r, g, b = t[1], t[2], t[3]
+		power:SetStatusBarAtlas(atlas)
+		power:SetStatusBarColor(1, 1, 1)
```

Comment:
Hope we aren't talking about `bg.multiplier` now 😁 

Anyways, here's my test setup:
```Lua
self:SetSize(238, 24)

local power = CreateFrame("StatusBar", nil, self)
power:SetAllPoints()
power:SetStatusBarTexture("Interface\\BUTTONS\\WHITE8X8")
power.colorPower = true
power.frequentUpdates = true
-- power.useAtlas = true

local bg = power:CreateTexture(nil, 'BACKGROUND')
bg:SetAllPoints(power)
bg:SetColorTexture(1, 1, 1, .4)
power.bg = bg

self.Power = power
```
Your current implementation breaks it at [#L118](https://github.com/oUF-wow/oUF/blob/cac1dec131163035591751ee3be2075e062ccb5c/elements/power.lua#L181), cuz `power.bg.texture` is `Color-ffffff80`, so I commented out that bg texture reset.

That's how it looks like:

![image](https://cloud.githubusercontent.com/assets/2725970/18806344/2696183a-8254-11e6-9398-24bd25b31202.png)

Then I enabled atlas:
![image](https://cloud.githubusercontent.com/assets/2725970/18806342/165a8f1e-8254-11e6-8ca9-2a0400d79851.png)

As you can see, cuz `bg.multiplier` is 1, there's nearly no difference between fg and bg textures.

Then I removed `bg:SetAtlas(atlas)` thingy:
![image](https://cloud.githubusercontent.com/assets/2725970/18806336/f8efa0a4-8253-11e6-9d1b-b1ff98bde22a.png)

Better, and you don't need to worry about bg being a texture or color-texture 😁 But `t` may be `nil`, that's why I suggested copying that block from [#L190-L203](https://github.com/oUF-wow/oUF/blob/cac1dec131163035591751ee3be2075e062ccb5c/elements/power.lua#L190-L203). But code duplication sucks, mb reimplement that block as `ResolvePowerColor` function? 😒 

Or you could replace `r, g, b = t[1], t[2], t[3]` with `r, g, b = 0, 0, 0`:
![image](https://cloud.githubusercontent.com/assets/2725970/18806335/e905a8dc-8253-11e6-9bae-bdce25a59be1.png)

This way you don't need to worry about anything.